File: smartadapter/src/main/java/smartadapter/viewholder/DraggableViewHolder.java
package smartadapter.viewholder;

/*
 * Created by Manne Öhlund on 2019-08-16.
 * Copyright (c) All rights reserved.
 */

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import smartadapter.widget.BasicDragAndDropExtension;

/**
 * Provides target view for draggable purposes.
 * Lets ItemTouchHelper handler such as {@link BasicDragAndDropExtension}
 * to bind {@link androidx.recyclerview.widget.ItemTouchHelper#startDrag(RecyclerView.ViewHolder)}.
 */
public interface DraggableViewHolder {

    /**
     * Returns a target draggable view
     * @return view
     */
    @NonNull
    View getDraggableView();
}

File: smartadapter/src/main/java/smartadapter/viewholder/DraggableViewHolder.kt
package smartadapter.viewholder

/*
 * Created by Manne Öhlund on 2019-08-16.
 * Copyright (c) All rights reserved.
 */

import android.view.View
import smartadapter.widget.BasicDragAndDropExtension

/**
 * Provides target view for draggable purposes.
 * Lets ItemTouchHelper handler such as [BasicDragAndDropExtension]
 * to bind [androidx.recyclerview.widget.ItemTouchHelper.startDrag].
 */
interface DraggableViewHolder {

    /**
     * Target draggable view
     */
    val draggableView: View
}
